### PR TITLE
Overhaul README.md, split out DEVELOPMENT.md

### PR DESCRIPTION
**Pull request by mnuttall**
Flesh out the README with more information on what the tool does and how to use it. 

This will have merge conflicts with https://github.com/rhd-gitops-example/services/pull/84 so @a-roberts or I are going to be unlucky :- /

### CODE REVIEW

**Comment by mnuttall** [comment]
Updated with merges for #84

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Notes for developers
+
+## Building
+
+You need Go version 1.14 to build this project.
```
Comment (bigkevmcd) [inline DEVELOPMENT.md:5]: Do we need 1.14 ?

I note that `go.mod` indicates that it requires 1.13?

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+## Testing
+
+Linting should be done first (this is done on Travis, and what's good locally should be good there too)
+
+Grab the linter if you haven't already: 
+
+```shell
+GO111MODULE=on go get github.com/golangci/golangci-lint/cmd/golangci-lint@v1.26.0
+```
```
Comment (bigkevmcd) [inline DEVELOPMENT.md:19]: If we're using Go 1.13+ do we need `GO111MODULE=on` given that the default `auto` does the right thing in almost all cases?

File: README.md [hunk 3]
```diff
@@ -1,96 +1,119 @@
...
+- `--cache-dir` : path on the local filesystem in which Git checkouts will be cached.
+- `--commit-email` : Git commits require an associated email address and username. This is the email address. May be set via ~/.gitconfig.
+- `--commit-message` : use this to override the commit message which will otherwise be generated automatically.
+- `--commit-name` : The other half of `commit-email`. Both must be set.
+- `--debug` : prints extra debug output if true.
+- `--from` : an https URL to a GitOps repository for 'remote' cases, or a path to a Git clone of a microservice for 'local' cases.
+- `--help`: prints the above text if true.
+- `--insecure-skip-verify` : skip TLS cerificate verification if true. Do not set this to true unless you know what you are doing.
+- `--keep-cache` : `cache-dir` is deleted unless this is set to true. Keeping the cache will often cause further promotion attempts to fail. This flag is mostly used along with `--debug` when investigating failure cases. 
+- `--repository-type` : the type of repository: github, gitlab or ghe (default "github")
```
Comment (bigkevmcd) [inline README.md:101]: How do you provide the URL for accessing the `ghe` installation?

File: DEVELOPMENT.md (hunk 2 above)
Comment (mnuttall) [reply DEVELOPMENT.md:19]: Trying this in a clean environment:
```
% docker run -it golang bash
root@2dd3d15955e1:/go# go get github.com/golangci/golangci-lint/cmd/golangci-lint@v1.26.0
go: cannot use path@version syntax in GOPATH mode
```
The command succeeds if prefixed with `GO111MODULE=on` so I'm going to say yes it's still needed.

File: README.md (hunk 3 above)
Comment (mnuttall) [reply README.md]: I've pushed a small addition to the `--repository-type` argument. The API endpoint is worked out for you in https://github.com/rhd-gitops-example/services/blob/d3bfd0f0701791aa689ca08dac46285708b24216/pkg/git/client.go#L19 - so `--from` and `--to` need to be of the same type.

File: DEVELOPMENT.md (hunk 1 above)
Comment (mnuttall) [reply DEVELOPMENT.md]: I've checked and go v1.13 is fine; I've updated the text.

File: DEVELOPMENT.md (hunk 2 above)
Comment (bigkevmcd) [reply DEVELOPMENT.md:19]: Ahhh...indeed, tho', maybe it should link to the installation instructions for golangci-lint ?

**Pull request merged**